lib/game.rb

```diff
@@ -0,0 +1,40 @@
+class Game
+  def initialize
+    @rolls = []
+  end
+
+  def roll(pins)
+    @rolls << pins
+  end
+
+  def score
+    result = 0
+    index = 0
+    10.times do
+      if @rolls[index] == 10
+        result += spare_result(index)
+        index += 1
+      elsif spare(index)
+        result += spare_result(index)
+        index += 2
+      elsif result += regular_result(index)
+        index += 2
+      end
+    end
+    result
+  end
+
+  private
+
```

Comment:
Not sure I fully understand the naming of each of these methods. Could it be sign posted any better?